packages/suite-desktop/src/support/DesktopUpdater.tsx

```diff
@@ -117,26 +118,29 @@ const DesktopUpdater = ({ setIsUpdateVisible }: DesktopUpdaterProps) => {
         return true;
     }, [desktopUpdate.window, desktopUpdate.state, desktopUpdate.latest]);
 
-    useEffect(() => setIsUpdateVisible(isVisible), [setIsUpdateVisible, isVisible]);
-
-    if (!isVisible) {
-        return null;
-    }
-
-    switch (desktopUpdate.state) {
-        case UpdateState.EarlyAccessEnable:
-            return <EarlyAccessEnable hideWindow={hideWindow} />;
-        case UpdateState.EarlyAccessDisable:
-            return <EarlyAccessDisable hideWindow={hideWindow} />;
-        case UpdateState.Available:
-            return <Available hideWindow={hideWindow} latest={desktopUpdate.latest} />;
-        case UpdateState.Downloading:
-            return <Downloading hideWindow={hideWindow} progress={desktopUpdate.progress} />;
-        case UpdateState.Ready:
-            return <Ready hideWindow={hideWindow} />;
-        default:
-            return null;
-    }
+    const getUpdateModal = () => {
+        switch (desktopUpdate.state) {
+            case UpdateState.EarlyAccessEnable:
+                return <EarlyAccessEnable hideWindow={hideWindow} />;
+            case UpdateState.EarlyAccessDisable:
+                return <EarlyAccessDisable hideWindow={hideWindow} />;
+            case UpdateState.Available:
+                return <Available hideWindow={hideWindow} latest={desktopUpdate.latest} />;
+            case UpdateState.Downloading:
+                return <Downloading hideWindow={hideWindow} progress={desktopUpdate.progress} />;
+            case UpdateState.Ready:
+                return <Ready hideWindow={hideWindow} />;
+            default:
+                return null;
+        }
+    };
+
+    return (
+        <>
+            {isVisible && getUpdateModal()}
+            <ModalContextProvider isDisabled={isVisible}>{children}</ModalContextProvider>
```

Comment:
But the _parent_ context shouldn't be disabled, only the _child_ context should. The `ModalContextProvider` has two main responsibilities:
1) to mark the area to which modals from its subtree should be rendered (with one global provider above all to be sure that all the modals could be rendered over the whole app, e.g. in `Main.tsx`), and
2) to tell other `ModalContextProvider`s nested in its subtree, that there is currently higher priority modal rendered somewhere above them (in this case update info), so they shouldn't be displayed. And it's not always this case, sometimes it's legit to render two modals over each other.

Modal decision-making could be described like this: Is the nearest `ModalContextProvider` above me disabled (either by its prop or by higher `ModalContextProvider` being disabled as well)?  If yes, don't show anything, otherwise render yourself into wrapping element defined by it.

Reply:
What I don't fully get is: the modal target is always coming from the closest context, so when there is a context inside a context the target from the top context is disregarded? In that case why not having a target separately, as I stated in a different comment, and have a fn from the context which could disable modals?

Reply:
Because the target from the top context could be still used from the "sibling" tree of the nested context.